module/zfs/dmu_recv.c

```diff
@@ -2761,6 +2765,22 @@ dmu_recv_end_sync(void *arg, dmu_tx_t *tx)
 		drc->drc_newsnapobj =
 		    dsl_dataset_phys(drc->drc_ds)->ds_prev_snap_obj;
 	}
+
+	/*
+	 * If this is a raw receive, the crypt_keydata nvlist will include
+	 * a to_vset_guid for us to set on the new snapshot. Older
+	 * implementations of the raw send code did not have this field.
+	 * In this case the ivset guid will be set to 0 which will disable
+	 * the check for this value.
```

Comment:
Is there a difference between having the value set to 0, and having the value not be present (as with older pools)?  If there's no difference, I think it would be better to only have one on-disk state to represent this.  I.e. we should not have a zero value, instead the fields should not be present.

Reply:
I agree. I'll make it not exist if the value is 0.